### ISSUE TITLE
Nullness Checker: error message when static field is not initialized

### ISSUE THREAD

**Issue by raduklb** [typetools/checker-framework#2415]
Hello,

Consider the following code:
```
import java.util.List;

public class Game {
	static List<Integer> assetsPile;

	public Game(List<Integer> assetsIds) {
		this.assetsPile = assetsIds;
	}
}

class Game2 {
	List<Integer> assetsPile2;

	public Game2(List<Integer> assetsIds2) {
		this.assetsPile2 = assetsIds2;
	}
}
```

Running `javac -g Game.java -processor nullness` issues:

> Game.java:3: error: [initialization.fields.uninitialized] the constructor does not initialize fields: assetsPile
> public class Game {
>        ^
> 1 error

which means that the problem occurs with static fields but not instance fields.

This [[CommandOutput.txt](https://github.com/typetools/checker-framework/files/3059811/CommandOutput.txt)] is the output when `javac -version -verbose -AprintAllQualifiers -g Game.java -processor nullness` command is run.

Best regards,
Radu Capalb

**Comment by mernst** [comment]
I think the bug here is how the error is reported, which is unnecessarily confusing.

The given code is wrong, so it is good that the Nullness Checker issues a warning.  For example, the following program prints "null":

```
public class TestGame {
  public static void main(String[] args) {
    System.out.println(Game.asset);
  }
}

class Game {
  static Integer asset;

  public Game(Integer asset) {
    this.asset = asset;
  }
}
```

The warning message should indicate that the problem is with the static initializer for the class, not the constructor per se.  There is a hint in that the warning message appears on the line with `class Game`, not on the constructor.

Thanks for pointing out this confusing message.
